packages/peregrine/src/hooks/__tests__/useDOMQuery.spec.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import {
+    TestValidComponent,
+    TestInvalidComponent
+} from '../__mocks__/useDOMQuery';
+
+describe('Unit testing useDOMQuery hook.', () => {
+    beforeAll(() => {
+        const div = document.createElement('div');
+        window.domNode = div;
+        document.body.appendChild(div);
+    });
+
+    it('setInnerText, setInnerHTML and setAttribute manipulate DOM when a valid query selector is provided', () => {
+        const newText = 'Changed after mount.';
+        const newHTML = '<i>Changed after mount.</i>';
+        const newName = 'newName';
+        const wrapper = mount(
+            <TestValidComponent
+                newText={newText}
+                newHTML={newHTML}
+                newName={newName}
+            />,
+            { attachTo: window.domNode }
+        );
+        return new Promise(resolve => {
+            setTimeout(() => {
+                resolve();
+            }, 0);
+        }).then(() => {
+            const div1 = wrapper.find('#div1').getDOMNode();
+            const div2 = wrapper.find('#div2').getDOMNode();
+            expect(div1.innerText).toBe(newText);
+            expect(div2.innerHTML).toBe(newHTML);
+            expect(div2.getAttribute('name')).toBe(newName);
```

Comment:
These assertions are hard to reason about because the logic is spread between two files.

Each unit test should test one distinct thing. In this case, it would be something like:

```js
// 1. Setup / prepare the test if needed

// 2. Call the function under test
const [, { setInnerText }] = useDOMQuery('#div1');
setInnerText(newText);

// 3. Make your assertions
expect(div1.innerText).toBe(newText);
```